src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -164,6 +165,90 @@ namespace winrt::Microsoft::Terminal::TerminalControl::implementation
         });
     }
 
+    // Method Description:
+    // - Create the SearchBoxControl object, and attach it
+    //   to the Terminal Control root
+    // Arguments:
+    // - <none>
+    // Return Value:
+    // - <none>
+    void TermControl::CreateSearchBoxControl()
+    {
+        if (!_searchBox)
+        {
+            SearchBoxControl searchBox;
+            searchBox.HorizontalAlignment(HorizontalAlignment::Right);
+            searchBox.VerticalAlignment(VerticalAlignment::Top);
+            // We need to make sure the searchbox does not overlap
+            // with the scroll bar
+            Thickness searchBoxPadding = { 0, 0, _scrollBar.ActualWidth(), 0 };
+            searchBox.Margin(searchBoxPadding);
+
+            _searchBox = searchBox;
+            _root.Children().Append(_searchBox);
+
+            // Event handlers
+            _searchBox.Search({ this, &TermControl::_Search });
+            _searchBox.CloseButtonClicked({ this, &TermControl::_CloseSearchBoxControl });
+        }
+
+        _searchBox.SetFocusOnTextbox();
+    }
+
+    // Method Description:
+    // - Search text in text buffer. This is triggered if the user click
+    //   search button or press enter.
+    // Arguments:
+    // - SearchBoxControl: not used
+    // - text: the text to search
+    // Return Value:
+    // - <none>
+    void TermControl::_Search(const SearchBoxControl&, winrt::hstring text)
+    {
+        if (text.size() == 0)
+        {
+            return;
+        }
+
+        const Search::Direction direction = _searchBox.GetGoForward() ?

```

Comment:
Winrt Typed event handler does not support Boolean type, we may need to create a new Winrt type for these arguments and this will be a big change, I would put it as a future consideration because we need to check-in the first version as soon as possible considering that people are OOF this months. Actually, I discussed this with Dustin.